### PR TITLE
migrate Atoms/Molecules > Tables to Elements > Tables

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-615

### LGTMs
- [x] Dev LGTM
- [x] Dev LGTM
- [x] Design LGTM
- [x] QE LGTM

### CODE REVIEW

**Comment by CITguy** [comment]
### Atoms > Tables (Before)
atoms-tables-before

### Molecules > Tables (Before)
molecules-tables-before


### Elements > Tables (After)
* first section from `Atoms > Tables*
* remaining sections from `Molecules > Tables`

tables-after

**Comment by CITguy** [comment]
### rxPermalink
I tweaked functionality to get permalinking functional again. The style of rxPermalink was inspired by how momentjs does theirs. @glynnis, I'd like to hear your feedback on the coloring/styles for rxPermalink.

rx-permalink

File: demo/scripts/demoApp.js [hunk 1]
```diff
@@ -409,10 +424,6 @@ angular.module('demoApp', ['encore.ui', 'ngRoute'])
     rxPageTitle.setSuffix(' - EncoreUI');
 
     $rootScope.$on('$routeChangeSuccess', function () {
-        if ($location.hash()) {
-            $anchorScroll();
-        } else {
-            $window.scrollTo(0,0);
-        }
+        $timeout($anchorScroll, 250);
```
Comment (CITguy) [inline demo/scripts/demoApp.js]: I found that if you navigate to a permalink directly, it wasn't scrolling correctly. We have to wait a tick for content to load before scrolling.

File: demo/scripts/rxPermalink.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        scope: {},
+        transclude: true,
+        template: '<a ng-href="{{permalinkUrl}}" class="permalink" id="{{permaId}}" ng-transclude></a>',
+        link: function (scope, el, attr) {
+            scope.permaId = el.text()
+                .toLowerCase() // normalize to lowercase
+                .replace(/[^0-9a-z]/gi, '-') // non-alphanumeric to hyphen
+                .replace(/\-{2,}/g, '-'); // condense consecutive hyphens
+
+            scope.permalinkUrl = '#' + $location.path() + '#' + scope.permaId;
```
Comment (CITguy) [inline demo/scripts/rxPermalink.js]: We have to use `$location.path()` or else we'd get weird urls when clicking a permalink at a location that is already permalinked (double hashes). This ensures that we exclude the present hash (if any) when building our new url string.

File: demo/scripts/rxPermalink.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    return {
+        restrict: 'E',
+        scope: {},
+        transclude: true,
+        template: '<a ng-href="{{permalinkUrl}}" class="permalink" id="{{permaId}}" ng-transclude></a>',
+        link: function (scope, el, attr) {
+            scope.permaId = el.text()
+                .toLowerCase() // normalize to lowercase
+                .replace(/[^0-9a-z]/gi, '-') // non-alphanumeric to hyphen
+                .replace(/\-{2,}/g, '-'); // condense consecutive hyphens
```
Comment (CITguy) [inline demo/scripts/rxPermalink.js]: convert text content to url-safe string

**Comment by CITguy** [comment]
I just realized that I still need to add keyword filtering on the All Modules page. DO NOT MERGE

**Comment by CITguy** [comment]
Ready to review.

### Keyword Filtering

allmodules-with-keyword-filtering

* Tables is the only module that matches "pagination", at present

File: demo/scripts/rxPermalink.js (hunk 2 above)
Comment (100stacks) [inline demo/scripts/rxPermalink.js]: @CITguy, nice work here with URL text conversion using regular expressions!

File: src/elements/Tables/Tables.meta.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+    "displayName": "Tables",
+    "stability": "unstable",
```
Comment (100stacks) [inline src/elements/Tables/Tables.meta.json]: @CITguy, just curious are `Tables` still considered unstable?  Do you foresee major changes to their usage/implementation?

**Comment by 100stacks** [comment]
LGTM

File: src/elements/Tables/Tables.meta.json (hunk 4 above)
Comment (CITguy) [inline src/elements/Tables/Tables.meta.json]: to be honest, the stability index seems to break down a bit with the combined Element page. There are pieces in relation to Tables that are experimental (rxFloatingHeader, rxBulkSelect, rxSelectFilter, etc.), unstable (rxSortableColumn, etc.), and stable (rxActionMenu, basic table CSS and striping, etc). 

_**Note:** not all mentioned components are currently categorized within Tables._

Comment (CITguy) [inline src/elements/Tables/Tables.meta.json]: FYI: stability index strategy should be addressed in a separate PR and considered out of scope for this PR.

Comment (100stacks) [inline src/elements/Tables/Tables.meta.json]: Okay,  added a JIRA story to capture discussion:
> https://jira.rax.io/browse/FRMW-795

**Comment by glynnis** [comment]
All the tables re-org/new IA stuff looks good to me! @CITguy and I talked about pulling the anchor link styles out into a separate PR so there's more time to adjust their appearance. Everything else here LGTM.

**Comment by parlarjb** [comment]
LGTM
